third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -3041,12 +3138,43 @@ impl<'a, 'b> serde::Serialize for AnnotatedValue<'a, 'b, MoveTypeLayout, ValueIm
                     )));
                 }
                 (AnnotatedValue {
+                    exchange: self.exchange,
                     layout: &MoveTypeLayout::Address,
                     val: &v[0],
                 })
                 .serialize(serializer)
             },
 
+            // Special case: aggregatable values.
+            (MoveTypeLayout::Aggregatable(layout), val) => {
+                match self.exchange {
+                    Some(exchange) => {
+                        // If values are supposed to be exchanged, first re-interpret
+                        // the current value as ID and then get the actual value.
+                        // Then continue with serialization on a new value.
+                        let id = val.try_as_identifier().map_err(serde::ser::Error::custom)?;
```

Comment:
let's copy the &value_impl explicitly and then apply the generic function. We can add a comment saying that the clone could get expensive for future use-cases.